src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCommon.java

```diff
@@ -0,0 +1,299 @@
+            Object[] msgArgs = {positionInSource};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        checkDuplicateColumnName(positionInSource, name);
+        switch (jdbcType) {
+            /*
+             * SQL Server supports numerous string literal formats for temporal types, hence sending them as varchar with approximate
+             * precision(length) needed to send supported string literals. string literal formats supported by temporal types are available in MSDN
+             * page on data types.
+             */
+            case java.sql.Types.DATE:
+            case java.sql.Types.TIME:
+            case java.sql.Types.TIMESTAMP:
+            case microsoft.sql.Types.DATETIMEOFFSET:
+                if (this instanceof SQLServerBulkCSVFileRecord) {
+                    columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));
+                }
+                else if (this instanceof SQLServerBulkBatchInsertRecord) {
+                    columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, precision, scale, dateTimeFormatter));
+                }
+                else {
+                    columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));
+                }
+                break;
+
+            // Redirect SQLXML as LONGNVARCHAR
+            // SQLXML is not valid type in TDS
+            case java.sql.Types.SQLXML:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.LONGNVARCHAR, precision, scale, dateTimeFormatter));
+                break;
+
+            // Redirecting Float as Double based on data type mapping
+            // https://msdn.microsoft.com/en-us/library/ms378878%28v=sql.110%29.aspx
+            case java.sql.Types.FLOAT:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.DOUBLE, precision, scale, dateTimeFormatter));
+                break;
+
+            // redirecting BOOLEAN as BIT
+            case java.sql.Types.BOOLEAN:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.BIT, precision, scale, dateTimeFormatter));
+                break;
+
+            default:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, precision, scale, dateTimeFormatter));
+        }
+
+        loggerExternal.exiting(loggerClassName, "addColumnMetadata");
+    }
+
+    /**
+     * Set the format for reading in dates from the batch/file.
+     * 
+     * @param dateTimeFormat
+     *            format to parse data sent as java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+     */
+    public void setTimestampWithTimezoneFormat(String dateTimeFormat) {
+        DriverJDBCVersion.checkSupportsJDBC42();
```

Comment:
No need for `DriverJDBCVersion.checkSupportsJDBC42();`

Reply:
done.